File: cwt-spring/src/test/java/com/cwtsite/cwt/domain/configuration/service/ConfigurationServiceTest.java
package com.cwtsite.cwt.domain.configuration.service;

import com.cwtsite.cwt.domain.configuration.entity.Configuration;
import com.cwtsite.cwt.domain.configuration.entity.enumeratuion.ConfigurationKey;
import org.junit.Assert;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.Mockito;
import org.mockito.junit.MockitoJUnitRunner;

import java.util.Optional;

@RunWith(MockitoJUnitRunner.class)
public class ConfigurationServiceTest {

    @InjectMocks
    private ConfigurationService configurationService;

    @Mock
    private ConfigurationRepository configurationRepository;

    @Test
    public void getParsedPointsPatternConfiguration() {
        Mockito
                .when(configurationRepository.findById(ConfigurationKey.POINTS_PATTERN))
                .thenReturn(Optional.of(createPointsPatternConfiguration("(3, 3), (2, 1)")))
                .thenReturn(Optional.of(createPointsPatternConfiguration("(10, 12),(0, 1),(2  , 1), (0,0),")));

        Assert.assertArrayEquals(
                new int[][]{new int[]{3, 3}, new int[]{2, 1}},
                configurationService.getParsedPointsPatternConfiguration().toArray());

        Assert.assertArrayEquals(
                new int[][]{
                        new int[]{10, 12},
                        new int[]{0, 1},
                        new int[]{2, 1}},
                configurationService.getParsedPointsPatternConfiguration().toArray());
    }

    private Configuration createPointsPatternConfiguration(String value) {
        final Configuration configuration = new Configuration();
        configuration.setValue(value);
        configuration.setKey(ConfigurationKey.POINTS_PATTERN);
        return configuration;
    }
}

File: cwt-spring/src/test/java/com/cwtsite/cwt/domain/configuration/service/ConfigurationServiceTest.kt
package com.cwtsite.cwt.domain.configuration.service

import com.cwtsite.cwt.domain.configuration.entity.Configuration
import com.cwtsite.cwt.domain.configuration.entity.enumeratuion.ConfigurationKey
import com.cwtsite.cwt.domain.tournament.entity.Tournament
import com.cwtsite.cwt.domain.tournament.service.TournamentService
import com.cwtsite.cwt.test.EntityDefaults
import com.cwtsite.cwt.test.MockitoUtils.anyObject
import org.assertj.core.api.Assertions.assertThat
import org.junit.runner.RunWith
import org.mockito.InjectMocks
import org.mockito.Mock
import org.mockito.Mockito.verify
import org.mockito.Mockito.`when`
import org.mockito.junit.MockitoJUnitRunner
import kotlin.test.Test

@RunWith(MockitoJUnitRunner::class)
class ConfigurationServiceTest {

    @InjectMocks private lateinit var cut: ConfigurationService
    @Mock private lateinit var configurationRepository: ConfigurationRepository
    @Mock private lateinit var tournamentService: TournamentService

    @Test
    fun save() {
        val tournament = EntityDefaults.tournament()
        `when`(tournamentService.getCurrentTournament()).thenReturn(tournament)
        val configuration = Configuration(
            key = ConfigurationKey.NUMBER_OF_GROUP_MEMBERS_ADVANCING,
            value = "2"
        )
        `when`(tournamentService.save(anyObject()))
            .thenAnswer {
                val value = Integer.parseInt(configuration.value)
                assertThat(it.getArgument<Tournament>(0).numOfGroupAdvancing).isEqualTo(value)
                tournament.copy(numOfGroupAdvancing = value)
            }
        cut.save(configuration)
        verify(tournamentService).save(tournament)
    }
}
